src/drive/web/modules/filelist/File.jsx

```diff
@@ -194,26 +196,46 @@ const _Size = ({ filesize = '-' }) => (
 
 const Size = React.memo(_Size)
 
-const Status = ({ isAvailableOffline, id }) => (
-  <div
-    className={classNames(
-      styles['fil-content-cell'],
-      styles['fil-content-status']
-    )}
-  >
-    {isAvailableOffline && (
-      <span className={styles['fil-content-offline']}>
-        <Icon
-          icon="phone-download"
-          color={palette.white}
-          width="14"
-          height="14"
+const Status = ({ isAvailableOffline, file }) => {
+  const { pushModal, popModal } = useContext(ModalContext)
+
+  return (
+    <div
+      className={classNames(
+        styles['fil-content-cell'],
+        styles['fil-content-status']
+      )}
+    >
+      {isAvailableOffline && (
+        <span className={styles['fil-content-offline']}>
+          <Icon
+            icon="phone-download"
+            color={palette.white}
+            width="14"
+            height="14"
+          />
+        </span>
+      )}
+      <HammerComponent
+        onClick={() => {
+          pushModal(
+            <ShareModal
+              document={file}
+              documentType="Files"
+              sharingDesc={file.name}
+              onClose={popModal}
+            />
```

Comment:
I do not think we should use this pattern of pushing / popping modals. It breaks the React flow since the ShareModal component will not be updated if `file` (or any of its props) changes.

Reply:
+1 on a rencontré le problème sur Contacts, avec le conclusion qu'on devrait refacto pour ne pas utiliser de pushModal(). Le souci était qu'on souhaitait avoir un spinner sur le bouton de la modale, ce qui nécessite donc un re-render de la modale.